scene/animation/animation_player.cpp

```diff
@@ -664,7 +664,7 @@ void AnimationPlayer::_animation_process_animation(AnimationData *p_anim, double
 					}
 				}
 
-				if (update_mode == Animation::UPDATE_CONTINUOUS || update_mode == Animation::UPDATE_CAPTURE || (p_delta == 0 && update_mode == Animation::UPDATE_DISCRETE)) { //delta == 0 means seek
+				if (update_mode == Animation::UPDATE_CONTINUOUS || update_mode == Animation::UPDATE_CAPTURE || update_mode == Animation::UPDATE_DISCRETE) {
```

Comment:
This section should only be processed at the moment of playback or seeking. With this change, this is always processed even though it is in DC mode, so this implementation is wrong. In my opinion, we probably need to implement a special case for DC mode when firing `play()`.

Reply:
Hi @TokageItLab ! I've added a custom section for DISCRETE mode, reusing most parts from the CAPTURE processing, but using `track_get_key_value`.

About this part:

```
if (pa->accum_pass != accum_pass) {
	ERR_CONTINUE(cache_update_prop_size >= NODE_CACHE_UPDATE_MAX);
	cache_update_prop[cache_update_prop_size++] = pa;
	pa->accum_pass = accum_pass;
}
pa->value_accum = first_value;
```

I am not sure what is the `accum_pass` responsible for, but not seeing any point in interpolating the value in the `else` clause, I've removed the `else` clause entirely. Now we assign `value_accum` always, outside of the `if` statement. Please let me know if this solution is ok :)